### PR TITLE
Scatter plot

### PR DESCRIPTION
**Pull request by VictoriaBeilsten-Edmands**

### CODE REVIEW

File: client/component/ScatterPlot.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import '@h5web/lib/dist/styles.css';
+import { ScatterVis } from '@h5web/lib';
+import ndarray from 'ndarray';
+import React from 'react';
+import type {TypedArray} from 'ndarray';
+
+
+
+type ScatterProps = {
```
Comment (PeterC-DLS) [inline client/component/ScatterPlot.tsx:9]: Delete

File: client/component/ScatterPlot.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  xData: number[];
+  yData: number[];
+  dataArray: ndarray.NdArray<TypedArray>;
+  domain: [number, number];
+  axesParameters: AxesParameters;
+};
+
+class ScatterPlot extends React.Component<ScatterProps> {
```
Comment (PeterC-DLS) [inline client/component/ScatterPlot.tsx:17]: Use `ScatterPlotProps`

File: client/component/ScatterPlot.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  render() {
+    return (
+      <>
+        <ScatterVis
+            abscissaParams={{value: this.props.xData}}
```
Comment (PeterC-DLS) [inline client/component/ScatterPlot.tsx:22]: Needs to convert using `Array.from()` with requested changes

File: client/component/PlotComponent.tsx [hunk 4]
```diff
@@ -347,6 +380,21 @@ class PlotComponent extends React.Component<PlotComponentProps, PlotStates> {
       );
     }
     }
+    if (this.state.scatterData !== undefined) {
+      const i = this.state.scatterData as ScatterData;
```
Comment (PeterC-DLS) [inline client/component/PlotComponent.tsx:384]: Should be `DScatterData` (fix other cases too)

File: client/component/main.d.ts [hunk 5]
```diff
@@ -73,6 +85,14 @@ interface LinePlotProps {
   axesParameters: AxesParameters;
 }
 
+interface ScatterPlotProps {
+  xData: number[];
```
[the remark this answers is not in the document]
Comment (VictoriaBeilsten-Edmands) [reply client/component/main.d.ts]: I made this one `NdArray<TypedArray>`

**Review by PeterC-DLS — APPROVED** [review]
Looks fine

**Pull request merged**